src/world/BallPossession.cpp

```diff
@@ -29,10 +28,13 @@ void BallPossession::recomputeState() {
     bool weAreFar = farFromUsTime > FAR_TIME_TRESHOLD;
     bool theyAreFar = farFromThemTime > FAR_TIME_TRESHOLD;
 
-    if ((weAreClose && ! theyAreClose) || (world::world->getBall()->pos.x > world::field->get_field().field_length()/4.0)) {
+    FieldMessage field = world::field->get_field();
+    double ourPossessionX = field.getLeftLineX() + OUR_POSSESSION_RELATIVE_WIDTH_THRESHOLD * field.field_width();
```

Comment:
Very shady indeed

Reply:
I now changed field_width to field_length (it was being updated in a newer version of the code which I did not commit), but it should not be 5/8th the field but 6/8th the field. Since the field length is measured from the left part of the field (negative x coordinate) to the right part of the field (positive x coordinate). So field_length/4.0 is half in between the middle line and the left line and you measure this difference from the middle line which is at 1/2 of the field_length. Therefore it should be 6/8 = 3/4 = 1/2 + 1/4